src/style-spec/validate_mapbox_api_supported.js

```diff
@@ -44,15 +45,22 @@ function getSourceErrors(source: Object, i: number): Array<?ValidationError> {
     const sourceKeys = ['type', 'url', 'tileSize'];
     errors.push(...getAllowedKeyErrors(source, sourceKeys, 'source'));
 
+    /*
+     * "type" is required and must be one of "vector", "raster", "raster-dem"
+     */
+    if (!acceptedSourceTypes.has(String(source.type))) {
```

Comment:
Need to use the String constructor here because gl js validation uses `@mapbox/jsonlint-lines-primitives` to parse strings into JSON, which turns primitives into objects with a line number property.

Reply:
Whoa fancy